src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,136 @@
+    
+    for row in data_csv:
+        if row[1] != "NULL":
+            SPARQL = '''
+            SELECT * WHERE {
+                SERVICE wikibase:mwapi {
+                    bd:serviceParam wikibase:api "EntitySearch" ;
+                                    wikibase:endpoint "www.wikidata.org" ;
+                                    mwapi:search "%(Title)s" ;
+                                    mwapi:language "en" .
+                    ?item wikibase:apiOutputItem mwapi:item .
+                }
+
+                ?item wdt:P31/wdt:P279* wd:Q11424 .
+                
+                {
+                    # Get US release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    ?releaseDateStatement pq:P291 wd:Q30 .  
+                }
+                UNION
+                {
+                    # Get unspecified release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+                }
+            
+                FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+                ?item rdfs:label ?itemLabel .
+                FILTER (lang(?itemLabel) = "en") .
+
+                OPTIONAL {
+                    ?item wdt:P136 ?genre .
+                    ?genre rdfs:label ?genreLabel .
+                    FILTER (lang(?genreLabel) = "en") .
+                }
+
+                OPTIONAL {?item wdt:P57 ?director.
+                                ?director rdfs:label ?directorLabel.
+                                FILTER (lang(?directorLabel) = "en")}
+
+                SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+                }
+        
+            ''' % {'Title': row[2], 'Year': int(row[1])}
+
+            response = requests.post('https://query.wikidata.org/sparql',
+                      headers={'User-Agent': user_agent},
+                      data={
+                        'query': SPARQL,
+                        'format': 'json',
+                      }
+        )
+            response.raise_for_status() 
+            
+            data = response.json()
+            
+            wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+            wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+            wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+        
+    return(wiki_movie_ids, wiki_genres, wiki_directors)
+
+# Calling all functions
+processed_data = []
+
+netflix_file = read_netflix_txt(base_dir + 'netflix_movies.txt')
+
+netflix_csv = base_dir + 'netflix_movies.csv'
+
+wiki_movie_ids_list, wiki_directors_list, wiki_genres_list = wiki_query(netflix_file, user_agent)
+
+for index, row in enumerate(netflix_file):
+    title, year, netflix_id = row[0], int(row[1]), row[2]
```

Comment:
Nit: You can just put `row` on the right side, and as long as the `row` list only has three elements, Python will destructure it for you.

```
title, year, netflix_id = row
```

Reply:
In fact, you can even put the tuple right in your for loop:

```
for index, (title, year, netflix_id) in enumerate(netflix_file):
    ...
```

Reply:
I can update this using the first solution; Kind of curious if you know why it doesn't raise an error for year / how it knows to treat row[1] as int? (because the initial solution explicitly converts str year to int year)

Reply:
You're right. Technically my suggestion doesn't actually map one to one with what is currently there. If you destructure the row, you will get 3 strings for title, year and netflix_id.

However I ignored the cast to `int` because it's unnecessary. Yes the year is an integer. But it's fine leaving it as a str because it's just being used to format the SPARQL query. So if it goes into the format as an int, it gets converted back to str and placed in the query as a string anyway.

Even when it is output, it is being written to a CSV, so that's also a string.

Reply:
Gotcha, thanks!